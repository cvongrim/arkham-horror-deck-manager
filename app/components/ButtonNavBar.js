// React Library Imports
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {Platform, StyleSheet, TouchableOpacity} from 'react-native';
import {Navigation} from 'react-native-navigation';
import Icon from 'react-native-vector-icons/MaterialIcons';

// Constants and Styles
import colors from '../styles/colors';

/**
 * Create custom buttons to display in the Nav bar
 */
class ButtonNavBar extends Component {
    /**
     * Constructor
     * @param {object} props
     */
    constructor(props) {
        super(props);
    }

    /**
     * Render the button
     * @return {*}
     */
    render() {
        let firstStyling = {};
        let osIconStyling = {};
        let osButtonContainerStyling = {};

        // Style the first button (Farthest Right) only on Android
        // If we don't add this, is will be flush with the edge of the screen
        if (this.props.first && Platform.OS === 'android') {
            firstStyling = {
                left: -5,
            };
        }

        // Android needs some special styling to center the buttons
        if (Platform.OS === 'android') {
            osIconStyling = {
                top: 5,
            };
            osButtonContainerStyling = {
                width: 50,
            };
        }

        return (
            <TouchableOpacity
                accessible={true}
                accessibilityLabel={'Open ' + this.props.label}
                hitSlop={{top: 20, bottom: 20, left: 20, right: 20}}
                onPress={() => Navigation.handleDeepLink({link: this.props.link})}
                style={[styles.buttonContainer, osButtonContainerStyling]}
            >
                <Icon
                    style={[styles.button, firstStyling, osIconStyling]}
                    color={'#ffffff'}
                    name={this.props.icon}
                    size={30}

                />
            </TouchableOpacity>
        );
    }
}

ButtonNavBar.propTypes = {
    first: PropTypes.bool,
    icon: PropTypes.string,
    link: PropTypes.string,
    label: PropTypes.string,
};

ButtonNavBar.defaultProps = {
    color: colors.black,
    first: false,
};

const styles = StyleSheet.create({
    button: {
        height: 30,
        width: 30,
    },
    buttonContainer: {
        alignItems: 'center',
        height: 40,
        justifyContent: 'center',
        width: 30,
    },
});

export default ButtonNavBar;
