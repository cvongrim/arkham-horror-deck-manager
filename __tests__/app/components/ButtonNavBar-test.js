/* global test expect */
import 'react-native';
import React from 'react';
import ButtonNavBar from '../../../app/components/ButtonNavBar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import colorStyles from '../../../app/styles/colors';

test('ButtonNavBar renders correctly', () => {
    const tree = renderer.create(<ButtonNavBar
        color={colorStyles.black}
        icon={'menu'}
        link={'menuScreen'}
        label={'Menu'}
        first={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});
