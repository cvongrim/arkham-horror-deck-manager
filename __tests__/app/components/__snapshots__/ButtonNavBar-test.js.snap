// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonNavBar renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel="Open Menu"
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 30,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 30,
        },
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          Object {},
          Object {},
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î—’
  </Text>
</View>
`;
